generate_kubernetes_enums.js

```diff
@@ -0,0 +1,145 @@
+            }
+        }
+    });
+    return {type: jsonType || type, name: jsonProp, field: name};
+}
+
+function parse_enum_types(types_go, swagger_json) {
+    const enumTypes = [];
+    const enumTypeList = {};
+    const typeList = {};
+
+    const fileData = types_go.replace(COMMENT_REGEX, '');
+    const lines = _.compact(_.map(fileData.split('\n'), _.trim));
+    for (let i = 0; i < lines.length; i++) {
+        let l = lines[i];
+        const m  = /type\s+(\w+)\s+string/g.exec(l.trim());
+        if (m) {
+            enumTypes.push(m[1]);
+            continue;
+        }
+        const type =  /type\s+(\w+)\s+struct\s{/g.exec(l);
+        if (type) {
+            i++;
+            typeList[type[1]] = [];
+            while (lines[i] !== '}') {
+                if (lines[i].includes('}')) {
+                    throw new Error('unexpected quote' + lines[i]);
+                }
+                const field = parse_protocol_buf(lines[i]);
+                if ((!field || !field.name) && lines[i] !== 'metav1.TypeMeta `json:",inline"`') {
+                    throw new Error(`invalid line ${lines[i]}`)
+                } else {
+                    if (enumTypes.indexOf(field.type) >= 0) {
+                       typeList[type[1]].push(field);
+                    }
+                }
+
+                i++;
+            }
+            continue;
+        }
+        if (/const\s+\(/g.exec(l)) {
+            i++;
+            while (lines[i] !== ')') {
+                if (lines[i].includes(')')) {
+                    throw new Error('unexpected quote' + lines[i]);
+                }
+                const n = /(\w+)\s+(\w+)\s+=\s+"(\w+)"/.exec(lines[i]);
+                if (n) {
+                    const typeName = n[2];
+                    if (typeName !== 'string') {
+                        if (enumTypes.indexOf(typeName)<0)  {
+                            throw new Error(`invalid type ${typeName}`);
+                        }
+                        if (!enumTypeList[typeName]) {
+                            enumTypeList[typeName] = [n[3]];
+                        } else {
+                            enumTypeList[typeName].push(n[3]);
+                        }
+                    }
+
+                } else {
+                    // ignore
+                }
+                i++;
+            }
+        }
+    }
+    const json = JSON.parse(swagger_json);
+    const map = {};
+
+    _.each(Object.keys(typeList), key => {
+        let t = typeList[key];
+        _.each(t, field => {
+            if (enumTypeList[field.type]) {
+                // console.log('Got:', key, field.name, enumTypeList[field.type].join('|'))
+                map[key + '!' + field.name] = {
+                    enums: enumTypeList[field.type]
+                }
+            }
+        });
+    });
+    let current;
+    const enums = {};
+    _.each(json.definitions, (definitions, key) => {
+        current = key.split('.').slice(-1)[0];
```

Comment:
Can't we `let current` here?  I thought the point of `let` was to allow block scoping.